specfile/changelog.py

```diff
@@ -0,0 +1,127 @@
+# Copyright Contributors to the Packit project.
+# SPDX-License-Identifier: MIT
+
+import collections
+import datetime
+import re
+from typing import List
+
+from specfile.sections import Section
+
+DAYS = [
+    "Mon",
+    "Tue",
+    "Wed",
+    "Thu",
+    "Fri",
+    "Sat",
+    "Sun",
+]
+
+MONTHS = [
+    "Jan",
+    "Feb",
+    "Mar",
+    "Apr",
+    "May",
+    "Jun",
+    "Jul",
+    "Aug",
+    "Sep",
+    "Oct",
+    "Nov",
+    "Dec",
+]
+
+
+class ChangelogEntry:
+    """
+     Class that represents a changelog entry.
+
+    Attributes:
+        date: Date of the entry.
+        author: Author of the entry.
+        evr: EVR (epoch, version, release) of the entry.
+        content: List of lines forming the content of the entry.
+    """
+
+    def __init__(
+        self, date: datetime.date, author: str, evr: str, content: List[str]
+    ) -> None:
+        self.date = date
+        self.author = author
+        self.evr = evr
+        self.content = content.copy()
+
+    def __str__(self) -> str:
+        return f"{self.header}\n" + "\n".join(self.content) + "\n"
+
+    def __repr__(self) -> str:
+        date = repr(self.date)
+        content = repr(self.content)
+        return f"ChangelogEntry({date}, '{self.author}', '{self.evr}', {content})"
+
+    @property
+    def header(self) -> str:
+        """Entry header in the format that is being used in Fedora spec files."""
+        weekday = DAYS[self.date.weekday()]
+        month = MONTHS[self.date.month - 1]
+        return f"* {weekday} {month} {self.date:%d %Y} {self.author} - {self.evr}"
```

Comment:
`rpmdev-bumpspec` uses time as well for me, e.g.:

```
 %changelog
* Mon Jan 24 14:13:29 CET 2022 Frantisek Lachman <flachman@redhat.com> - 0.5.3-1
- New upstream release 0.5.3
```

Reply:
You got me there, I didn't realize RPM supports this (since [4.14.0](https://github.com/rpm-software-management/rpm/commit/57f94a582602f0353cdb17a02dc12c4461d4f32d)). So we should definitely default to that.